packages/flutter_tools/lib/src/reporting/usage.dart

```diff
@@ -274,12 +272,12 @@ class _DefaultUsage implements Usage {
     _analytics.analyticsOpt = AnalyticsOpt.optOut;
   }
 
-  Analytics _analytics;
-  final FirstRunMessenger firstRunMessenger;
+  late Analytics _analytics;
```

Comment:
Is there a way we could avoid late here? Perhaps instead of a constructor that sets these in the body, we could have a static method or factory constructor for initialization

Reply:
I changed it to a static method.  I'm not sure it's better, the runtime null check just moved out of the property and into that method.  There's meandering logical paths to set up analytics and at some point we are pretty sure it's not null, so to keep the property nonnull it gets checked at runtime.  Let me know what you think, I'm open to suggestions.